### DISCUSSION TITLE
make an object active/deative

### DISCUSSION THREAD

**Discussion by LoveBumblebee** [bepu/bepuphysics2#119] [Q&A]
What is the cheapest way to make a dynamic object active/deative?(not awake/sleep)
1. use INarrowPhaseCallbacks.AllowContactGeneration
2. set a far away position and sleep (deactive),restore positon and awake (active)
3. remove and add again
Or is there another cheapest  way?

**Comment by RossNordby** [comment]
If you just want a body to use very little resources and not collide with anything, then probably just remove it from the simulation and readd it later.

Making callbacks to ignore it would require a bit of effort, and moving things really far away and sleeping them would both cost more immediately (sleeping moves just as much memory around as removing a body from the simulation) and it would make the broadphase have to do more topological updates to deal with the long distance teleport.
